extractor/filesystem/containers/k8simage/k8simage.go

```diff
@@ -0,0 +1,273 @@
+	Template       *PodTemplate `yaml:"template,omitempty"`
+	JobTemplate    *JobTemplate `yaml:"jobTemplate,omitempty"`
+}
+
+// JobTemplate represents a job template in CronJob resources.
+type JobTemplate struct {
+	Spec *JobSpec `yaml:"spec,omitempty"`
+}
+
+// JobSpec represents the spec of a Job.
+type JobSpec struct {
+	Template *PodTemplate `yaml:"template,omitempty"`
+}
+
+// PodTemplate represents a pod template in Kubernetes resources.
+type PodTemplate struct {
+	Spec *PodSpec `yaml:"spec,omitempty"`
+}
+
+// PodSpec represents a pod specification.
+type PodSpec struct {
+	Containers     []Container `yaml:"containers,omitempty"`
+	InitContainers []Container `yaml:"initContainers,omitempty"`
+}
+
+// Container represents a container specification in Kubernetes.
+type Container struct {
+	Image string `yaml:"image"`
+}
+
+// Config is the configuration for the Extractor.
+type Config struct {
+	// Stats is a stats collector for reporting metrics.
+	Stats stats.Collector
+	// MaxFileSizeBytes is the maximum file size this extractor will unmarshal. If
+	// `FileRequired` receives a larger file, it will return false.
+	MaxFileSizeBytes int64
+}
+
+// DefaultConfig returns the default configuration for the extractor.
+func DefaultConfig() Config {
+	return Config{
+		MaxFileSizeBytes: DefaultMaxFileSizeBytes,
+	}
+}
+
+// Extractor extracts container image references from Kubernetes YAML files.
+type Extractor struct {
+	stats            stats.Collector
+	maxFileSizeBytes int64
+}
+
+// New returns a Kubernetes container image extractor.
+//
+// For most use cases, initialize with:
+// ```
+// e := New(DefaultConfig())
+// ```
+func New(cfg Config) *Extractor {
+	return &Extractor{
+		stats:            cfg.Stats,
+		maxFileSizeBytes: cfg.MaxFileSizeBytes,
+	}
+}
+
+// NewDefault returns an extractor with the default config settings.
+func NewDefault() filesystem.Extractor { return New(DefaultConfig()) }
+
+// Name of the extractor.
+func (e Extractor) Name() string { return Name }
+
+// Version of the extractor.
+func (e Extractor) Version() int { return 0 }
+
+// Requirements of the extractor.
+func (e Extractor) Requirements() *plugin.Capabilities { return &plugin.Capabilities{} }
+
+// FileRequired returns true if the specified file looks like a Kubernetes YAML file.
+// It determines if the specified file is a Kubernetes YAML file that should be processed
+// by checking the file extension (.yaml or .yml).
+func (e Extractor) FileRequired(api filesystem.FileAPI) bool {
+	// Only consider YAML/YML files
+	path := api.Path()
+	ext := strings.ToLower(filepath.Ext(filepath.Base(path)))
+	return ext == ".yaml" || ext == ".yml"
+}
+
+// Extract extracts container image references from a K8s configuration file.
+func (e Extractor) Extract(ctx context.Context, input *filesystem.ScanInput) (inventory.Inventory, error) {
+	// Defer context cancellation check
+	defer func() {
+		if err := ctx.Err(); err != nil {
```

Comment:
Should be fine to remove this - there's a separate ctx handling where Extract() is called so we only need context cancellation inside busy loops in Extract()